sdks/js-sdk/README.md

```diff
@@ -0,0 +1,150 @@
+
+## Problem Description
+
+The 3 target platforms (Cordova, React Native & Web) all support JavaScript as the development language. To develop Mobile Apps that integrate into Mobile Services, it is necessary to provide SDKs for Mobile Developers for the target platform & language. These SDKs need to be written & maintained. 
+
+## Proposed solution
+
+Provide JavaScript implementations (via a JavaScript SDK). This means the default way of implementing an SDK interface will be in JavaScript. However, this may not always be possible. 
+For the Cordova and React Native platforms we will provide a plugin that encapsulates the native code. This plugin conforms to an interface defined in the JS SDK that abstracts all the native platform specific code. Doing this will make Cordova, Web and React Native SDK's more consistent. We will bundle an
+implementation of that Interface for the Web Platform in the JS SDK. The plugins for Cordova and React Native will be separate repositories. The JS SDK requires the users to install the plugin for those Platforms and it will either try to auto-detect the platform, or require the user to pass in a `PlatformImpls` module when a service is instantiated.
+
+### Advantages
+
+- Ability to reuse code for Cordova, React Native & Web and work in a single repository
+- Can reuse JavaScript implementations for keycloak, push, sync.js etc.
+- No need to adjust native code (from Native SDKs) for plugin use. The native wrappers will not depend on the Android or iOS SDKs but provide their own implementations.
+
+## Architecture
+
+### Config parsing 
+
+Despite the built-in `JSON` API to parse the configuration file, the Core SDK will contain a dedicated module to parse the configuration file, this has the following benefits:
+
+- Better documentation of the configuration values via TypeScript interfaces
+- Encapsulation of changes in the config file structure
+- Possible support for dynamic and asynchronous fetching of future configuration values
+
+### Metrics
+
+As with the other SDKs, Core will contain the implementation for metrics gathering and reporting, providing a JS interface for uploading records which must also be used even by modules that rely heavily on native implementations, such as some of the Self-Defence Security Checks.
+
+The uploading of metrics records utilizes the networking layer provided by Core.
+
+### Individual SDKs
+
+Depending on the SDK, a TypeScript/JavaScript implementation may be possible. In cases where it isn't, existing plugins can be used, or new plugins can be created for services.
+
+## Adding a module to the JS SDK
+
+New modules written JavaScript or Typescript will have the following items:
+
+- A single `index` entry file that organizes and exposes all the exported contents of the module.
+- Manually written or compiled files for a Cordova plugin should be present in the `www/` folder by convention
```

Comment:
I think this is outdated and contradictory with the next paragraph. It's saying here where to put Cordova code inside a module but below it says the plugin is should be in a separated plugin, IMO the correct one is the latter. Or am I missing something?

Reply:
lets leave it